src/Microsoft.IdentityModel.Xml/DsigSerializer.cs

```diff
@@ -110,27 +111,41 @@ public virtual KeyInfo ReadKeyInfo(XmlReader reader)
                     else if (reader.IsStartElement(XmlSignatureConstants.Elements.KeyValue, XmlSignatureConstants.Namespace))
                     {
                         reader.ReadStartElement(XmlSignatureConstants.Elements.KeyValue, XmlSignatureConstants.Namespace);
-                            if (reader.IsStartElement(XmlSignatureConstants.Elements.RSAKeyValue, XmlSignatureConstants.Namespace))
-                            {
-                                // Multiple RSAKeyValues were found
-                                if (keyInfo.RSAKeyValue != null)
-                                    throw XmlUtil.LogReadException(LogMessages.IDX30015, XmlSignatureConstants.Elements.RSAKeyValue);
+                        if (reader.IsStartElement(XmlSignatureConstants.Elements.RSAKeyValue, XmlSignatureConstants.Namespace))
+                        {
+                            // Multiple RSAKeyValues were found
+                            if (keyInfo.RSAKeyValue != null)
+                                throw XmlUtil.LogReadException(LogMessages.IDX30015, XmlSignatureConstants.Elements.RSAKeyValue);
 
-                                keyInfo.RSAKeyValue = ReadRSAKeyValue(reader);
+                            keyInfo.RSAKeyValue = ReadRSAKeyValue(reader);
+                        }
+                        else
+                        {
+                            // Skip the element since it is not an <RSAKeyValue>
+                            if (LogHelper.IsEnabled(EventLogLevel.Warning))
+                            {
+                                LogHelper.LogWarning(LogMessages.IDX30300, reader.ReadOuterXml());
                             }
                             else
                             {
-                                // Skip the element since it is not an <RSAKeyValue>
-                                LogHelper.LogWarning(LogMessages.IDX30300, reader.ReadOuterXml());
+                                reader.Skip();
                             }
```

Comment:
looks like changed behavior. @keegan-caruso @brentschmaltz @jennyf19 assuming no tests are failing, however would be good to assess this change for any potential impact (perhaps not relevant as the reader.ReadEndElement does this implicitly already.

Same comment below.